tests/test_db.rs

```diff
@@ -528,6 +529,28 @@ fn test_open_with_ttl() {
     assert!(db.get(b"key1").unwrap().is_none());
 }
 
+#[test]
+fn test_open_as_single_threaded() {
+    let primary_path = DBPath::new("_rust_rocksdb_test_open_as_single_threaded");
+
+    let mut db = DbWithThreadMode::<SingleThreaded>::open_default(&primary_path).unwrap();
+    let db_ref1 = &mut db;
+    let opts = Options::default();
+    db_ref1.create_cf("cf1", &opts).unwrap();
+}
+
+#[test]
+fn test_open_as_multi_threaded() {
+    let primary_path = DBPath::new("_rust_rocksdb_test_open_as_multi_threaded");
+
+    let db = DbWithThreadMode::<MultiThreaded>::open_default(&primary_path).unwrap();
```

Comment:
After I wrote this, I realized there is some other compile-error message checks. ;) Maybe, should I write single thread version of this using that facility?